### PR TITLE
Fix: NewGRF feature cleanup.

### PR DESCRIPTION
**Pull request by PeterN**
* NewGRF features are documented in hex, so display as hex
* Crash when loading NewGRF with out of range feature.

### CODE REVIEW

**Comment by PeterN** [comment]
Possible issue, unknown features within a NewGRF get ignored, rather than disabling the file, so it effectively silently fails (unless you had debugging on). It may be preferable to disable the file.

**Comment by planetmaker** [comment]
> Possible issue, unknown features within a NewGRF get ignored, rather than disabling the file, so it effectively silently fails (unless you had debugging on). It may be preferable to disable the file.

That is more a feature IMHO than an issue / bug as it makes OpenTTD (somewhat) forward compatible with future NewGRFs

**Comment by PeterN** [comment]
That is also what I was thinking. It's certainly better than OpenTTD hard-crashing :-)

File: src/newgrf.cpp [hunk 1]
```diff
@@ -2103,7 +2103,7 @@ static ChangeInfoResult CanalChangeInfo(uint id, int numinfo, int prop, ByteRead
 	ChangeInfoResult ret = CIR_SUCCESS;
 
 	if (id + numinfo > CF_END) {
-		grfmsg(1, "CanalChangeInfo: Canal feature %u is invalid, max %u, ignoring", id + numinfo, CF_END);
+		grfmsg(1, "CanalChangeInfo: Canal feature 0x%02X is invalid, max %u, ignoring", id + numinfo, CF_END);
```
Comment (LordAro) [inline src/newgrf.cpp:2106]: Should be able to use `%#02X` to get the 0x prefix

Comment (PeterN) [reply src/newgrf.cpp:2106]: %#02X will output 0XAA instead of 0xAA. But actually this one isn't a NewGRF feature, so...

Comment (LordAro) [reply src/newgrf.cpp:2106]: mm, documentation seems to suggest `0X` or `0x`, and being consistent is probably better

**Review by LordAro — APPROVED** [review]

**Pull request merged**